etna/transforms/base.py

```diff
@@ -1,20 +1,182 @@
 from abc import ABC
 from abc import abstractmethod
 from copy import deepcopy
+from typing import List
+from typing import Union
 
 import pandas as pd
+from typing_extensions import Literal
 
 from etna.core import BaseMixin
+from etna.datasets import TSDataset
 
 
 class FutureMixin:
     """Mixin for transforms that can convert non-regressor column to a regressor one."""
 
 
-class DymmyInColumnMixin:
-    """Mixin for transforms that has no explicit in_column."""
+class NewTransform(ABC, BaseMixin):
+    """Base class to create any transforms to apply to data."""
+
+    def __init__(self, in_column: Union[Literal["all"], List[str], str] = "target"):
+        self.in_column = in_column
+
+    def get_regressors_info(self) -> List[str]:
+        """Return the list with regressors created by the transform.
+
+        Returns
+        -------
+        :
+            List with regressors created by the transform.
+        """
+        return []
+
+    @property
+    def required_features(self) -> Union[Literal["all"], List[str]]:
+        """Get the list of required features."""
+        required_features = self.in_column
+        if isinstance(required_features, list):
+            return required_features
+        elif isinstance(required_features, str):
+            if required_features == "all":
+                return "all"
+            return [required_features]
+        else:
+            raise ValueError("in_column attribute is in incorrect format!")
+
+    def _update_dataset(self, ts: TSDataset, df: pd.DataFrame, df_transformed: pd.DataFrame) -> TSDataset:
+        """Update TSDataset based on the difference between dfs."""
+        columns_before = set(df.columns.get_level_values("feature"))
+        columns_after = set(df_transformed.columns.get_level_values("feature"))
+
+        # Transforms now can only remove or only add/update columns
+        removed_features = list(columns_before - columns_after)
+        if len(removed_features) != 0:
+            ts.remove_features(features=removed_features)
+        else:
+            new_regressors = self.get_regressors_info()
+            ts.update_columns_from_pandas(df=df_transformed, regressors=new_regressors)
+        return ts
+
+    @abstractmethod
+    def _fit(self, df: pd.DataFrame) -> "NewTransform":
```

Comment:
Why do we need `_fit` to return itself? We don't use it in `fit`.

Reply:
Fixed